lib/amplitude.dart

```diff
@@ -12,20 +12,36 @@ import 'package:amplitude_flutter/events/group_identify_event.dart';
 class Amplitude {
   Configuration configuration;
   MethodChannel _channel = const MethodChannel("amplitude_flutter");
+  /// Whether the Amplitude instance has been successfully initialized
+  ///
+  /// ```
+  /// var amplitude = Amplitude(Configuration(apiKey: "apiKey"));
+  /// // If care about init complete
+  /// await amplitude.isBuilt;
+  /// ```
+  late Future<bool> isBuilt;
 
   /// Returns an Amplitude instance
-  Amplitude(this.configuration);
-
-  /// Initializes an Amplitude instance
   ///
   /// ```
   /// var amplitude = Amplitude(Configuration(apiKey: "apiKey"));
-  /// await amplitude.init();
+  /// // If care about init complete
+  /// await amplitude.isBuilt;
   /// ```
-  Future<void> init([MethodChannel? methodChannel]) async {
+  Amplitude(this.configuration, [MethodChannel? methodChannel]){
     _channel = methodChannel ?? this._channel;
-    return await _channel.invokeMethod(
-        "init", this.configuration.toMap());
+    isBuilt = _init();
```

Comment:
Is `isBuilt` idempotent or will it rerun init each time?

Reply:
`isBuilt` is `Future<Amplitude>` type. It will get the result of `_init()`. `_init()` will only run once. Calling `isBuilt` multiple times will just read the value but not rerun `_init()`.